server/s3_evbuffer_wrapper.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+#pragma once
+
+#ifndef __S3_SERVER_S3_EVBUFFER_WRAPPER_H__
+#define __S3_SERVER_S3_EVBUFFER_WRAPPER_H__
+
+#include <evhtp.h>
+#include <string>
+
+// Usage example:
+//  Say for 32mb read operation with 1 mb unit size object.
+//  S3Evbuffer *evbuf = new S3Evbuffer(32 * 1048576/* 32mb */, 1048576 /* 1mb */
+// )
+//  buf->to_motr_read_buffers(rw_ctx)
+//  ... motr will fill data in rw_ctx with async read op ...
+//  After read op success follow next steps.
+//  evbuf->commit_buffers()
+//  struct evbuffer *data = buf->release_ownership();
+//  delete evbuf;
+//  ...
+//  evhtp_obj->http_send_reply_body(ev_req, evbuf);
```

Comment:
Not sure here, is it a misprint?  In line 38 `evbuf` is `delete`d.  Did you mean to put `data` as last arg here?